include/rocksdb/db.h

```diff
@@ -2014,6 +2015,19 @@ class DB {
       ColumnFamilyHandle* column_family, const Range* range, std::size_t n,
       TablePropertiesCollection* props) = 0;
 
+  // Get the aggregated data age info per level. Data age info is only available
+  // when DB's time tracking is enabled and data's age info tracking is enabled.
+  // The former is enabled with `preserve_internal_time_seconds` and/or
+  // `preclude_last_level_data_seconds`. The latter is enabled via
+  // `CompactForTieringCollectorFactory`.
+  virtual Status GetDataCollectionAgeInfoForLevels(
```

Comment:
This seems a little niche to go in DB API. If we had a version of GetPropertiesOfAllTables that divided the data by level (rather than cross-refing with GetLiveFilesMetaData) then collating the data could be a utility function outside of DB.

Reply:
That's fair. I had a similar version that divide the functionality into a table property collecting API and a util API to convert that into data age. I was thinking not many other usage expect table properties per level. But that is a more general API. I think Zippy folks should be fine with it too. Thanks for the suggestion!